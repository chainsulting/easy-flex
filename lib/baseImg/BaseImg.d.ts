import React, { ImgHTMLAttributes } from 'react';
import { AbsoluteSize, BorderRadius, Distance, Height, ObjectFit, Size, Width } from '../types';
export interface BaseImgProps extends ImgHTMLAttributes<HTMLImageElement> {
    /** Component's border radius. */
    borderRadius?: BorderRadius | AbsoluteSize;
    /** Component's height. */
    height?: Height | Size;
    /** Component's margin of all sides. */
    margin?: Distance | AbsoluteSize;
    /** Component's bottom margin. */
    marginBottom?: Distance | AbsoluteSize;
    /** Component's left margin. */
    marginLeft?: Distance | AbsoluteSize;
    /** Component's right margin. */
    marginRight?: Distance | AbsoluteSize;
    /** Component's top margin. */
    marginTop?: Distance | AbsoluteSize;
    /** Component's left and right margin. */
    marginHorizontal?: Distance | AbsoluteSize;
    /** Component's top and bottom margin. */
    marginVertical?: Distance | AbsoluteSize;
    objectFit?: ObjectFit;
    round?: boolean;
    /** Component's width. */
    width?: Width | Size;
}
export declare const BaseImg: React.ForwardRefExoticComponent<BaseImgProps & React.RefAttributes<HTMLImageElement>>;
