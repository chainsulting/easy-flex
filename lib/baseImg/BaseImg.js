"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseImg = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var react_1 = require("react");
var styled_components_1 = __importDefault(require("styled-components"));
var border_1 = require("../utils/border");
var margin_1 = require("../utils/margin");
var size_1 = require("../utils/size");
var StyledBaseImg = styled_components_1.default.img(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\tbox-sizing: border-box;\n\tdisplay: flex;\n\tobject-fit: ", ";\n\t", "\n\t", "\n\t", "\n"], ["\n\tbox-sizing: border-box;\n\tdisplay: flex;\n\tobject-fit: ", ";\n\t", "\n\t", "\n\t", "\n"])), function (_a) {
    var objectFit = _a["data-object-fit"];
    return objectFit;
}, border_1.borderStyle, margin_1.marginStyle, size_1.sizeStyle);
exports.BaseImg = (0, react_1.forwardRef)(function (_a, ref) {
    var borderColor = _a.borderColor, borderRadius = _a.borderRadius, borderStyle = _a.borderStyle, borderWidth = _a.borderWidth, fullHeight = _a.fullHeight, fullWidth = _a.fullWidth, height = _a.height, margin = _a.margin, marginBottom = _a.marginBottom, marginHorizontal = _a.marginHorizontal, marginLeft = _a.marginLeft, marginRight = _a.marginRight, marginTop = _a.marginTop, marginVertical = _a.marginVertical, maxHeight = _a.maxHeight, maxWidth = _a.maxWidth, minHeight = _a.minHeight, minWidth = _a.minWidth, objectFit = _a.objectFit, round = _a.round, width = _a.width, props = __rest(_a, ["borderColor", "borderRadius", "borderStyle", "borderWidth", "fullHeight", "fullWidth", "height", "margin", "marginBottom", "marginHorizontal", "marginLeft", "marginRight", "marginTop", "marginVertical", "maxHeight", "maxWidth", "minHeight", "minWidth", "objectFit", "round", "width"]);
    var borderStyleProps = (0, border_1.useBorderStyleProps)({ borderColor: borderColor, borderRadius: borderRadius, borderStyle: borderStyle, borderWidth: borderWidth, round: round });
    var marginStyleProps = (0, margin_1.useMarginStyleProps)({
        margin: margin,
        marginBottom: marginBottom,
        marginHorizontal: marginHorizontal,
        marginLeft: marginLeft,
        marginRight: marginRight,
        marginTop: marginTop,
        marginVertical: marginVertical,
    });
    var sizeStyleProps = (0, size_1.useSizeStyleProps)({
        fullHeight: fullHeight,
        fullWidth: fullWidth,
        height: height,
        maxHeight: maxHeight,
        maxWidth: maxWidth,
        minHeight: minHeight,
        minWidth: minWidth,
        width: width,
    });
    return ((0, jsx_runtime_1.jsx)(StyledBaseImg, __assign({ "data-object-fit": objectFit }, borderStyleProps, marginStyleProps, sizeStyleProps, { ref: ref }, props)));
});
exports.BaseImg.displayName = 'BaseImg';
var templateObject_1;
