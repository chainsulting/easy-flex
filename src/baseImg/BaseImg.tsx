import React, { forwardRef, ImgHTMLAttributes } from 'react';
import styled from 'styled-components';
import { ObjectFit } from '../types';
import { BorderProps, borderStyle, BorderStyleProps, useBorderStyleProps } from '../utils/border';
import { MarginProps, marginStyle, MarginStyleProps, useMarginStyleProps } from '../utils/margin';
import { SizeProps, sizeStyle, SizeStyleProps, useSizeStyleProps } from '../utils/size';

const StyledBaseImg = styled.img<
	{
		'data-object-fit'?: ObjectFit;
	} & BorderStyleProps &
		MarginStyleProps &
		SizeStyleProps
>`
	box-sizing: border-box;
	display: flex;
	object-fit: ${({ 'data-object-fit': objectFit }) => objectFit};
	${borderStyle}
	${marginStyle}
	${sizeStyle}
`;

export type BaseImgProps = ImgHTMLAttributes<HTMLImageElement> &
	BorderProps &
	MarginProps &
	SizeProps & {
		objectFit?: ObjectFit;
	};

export const BaseImg = forwardRef<HTMLImageElement, BaseImgProps>(
	(
		{
			borderColor,
			borderRadius,
			borderStyle,
			borderWidth,
			fullHeight,
			fullWidth,
			height,
			margin,
			marginBottom,
			marginHorizontal,
			marginLeft,
			marginRight,
			marginTop,
			marginVertical,
			maxHeight,
			maxWidth,
			minHeight,
			minWidth,
			objectFit,
			round,
			width,
			...props
		},
		ref
	) => {
		const borderStyleProps = useBorderStyleProps({ borderColor, borderRadius, borderStyle, borderWidth, round });

		const marginStyleProps = useMarginStyleProps({
			margin,
			marginBottom,
			marginHorizontal,
			marginLeft,
			marginRight,
			marginTop,
			marginVertical,
		});

		const sizeStyleProps = useSizeStyleProps({
			fullHeight,
			fullWidth,
			height,
			maxHeight,
			maxWidth,
			minHeight,
			minWidth,
			width,
		});

		return (
			<StyledBaseImg
				data-object-fit={objectFit}
				{...borderStyleProps}
				{...marginStyleProps}
				{...sizeStyleProps}
				ref={ref}
				{...props}
			/>
		);
	}
);

BaseImg.displayName = 'BaseImg';
