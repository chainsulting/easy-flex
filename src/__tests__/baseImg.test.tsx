import { render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';
import { BaseImg } from '../baseImg/BaseImg';

describe('<BaseImg />', () => {
	test('should have the basic css settings', async () => {
		const result = render(<BaseImg />);
		expect(result.baseElement).toMatchSnapshot();
	});
});
