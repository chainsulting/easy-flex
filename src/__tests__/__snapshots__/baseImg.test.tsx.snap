// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseImg /> should have the basic css settings 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
}

<body>
  <div>
    <img
      class="c0"
      data-margin-bottom="0px"
      data-margin-left="0px"
      data-margin-right="0px"
      data-margin-top="0px"
    />
  </div>
</body>
`;
